src/Equinox.Cosmos/Cosmos.fs

```diff
@@ -79,8 +79,19 @@ type [<NoEquality; NoComparison>] // TODO for STJ v5: All fields required unless
 type JsonCompressedBase64Converter() =
     inherit JsonConverter<JsonElement>()
 
+    static member Compress (value: JsonElement) =
+        if value.ValueKind = JsonValueKind.Null || value.ValueKind = JsonValueKind.Undefined then
+            value
+        else
+            let input = System.Text.Encoding.UTF8.GetBytes(value.GetRawText())
+            use output = new MemoryStream()
+            use compressor = new System.IO.Compression.DeflateStream(output, System.IO.Compression.CompressionLevel.Optimal)
+            compressor.Write(input, 0, input.Length)
+            compressor.Close()
+            JsonDocument.Parse("\"" + System.Convert.ToBase64String(output.ToArray()) + "\"").RootElement
```

Comment:
ouch, feel those allocations ! 
is JsonDocument.Parse definitely the most direct way to generate a JsonElement with a string body?

Reply:
Unfortunately, at the current time, that's the best (and only) way to do it. I think that in 5.0, they're planning to make JsonElements easier to create. When that happens (I think ETA is Nov), we can switch this logic around a bit.

Reply:
ok - I guess we'll worry about the string concat allocations at that point!